homeassistant/components/device_tracker/config_entry.py

```diff
@@ -180,8 +188,12 @@ def battery_level(self) -> int | None:
         return None
 
     @property
-    def source_type(self) -> SourceType | str:
+    def source_type(self) -> SourceType:
```

Comment:
This change, although just a type annotation, may need a dev blog.

Reply:
> This change, although just a type annotation, may need a dev blog.

https://github.com/home-assistant/developers.home-assistant/pull/2333

Reply:
Note: shorthand attributes moved to a separate PR: #126599